From acb5ff4c73ac4d52201d7c421f488e2ead5c8b9c Mon Sep 17 00:00:00 2001
From: Jurij Smakov <jurij@wooyd.org>
Date: Sat, 11 Nov 2006 14:09:15 +0200
Subject: [PATCH] ffs: handle 0 argument (bug #8968)

Handle an argument of 0 in ffs(), instead of looping indefinitely.
Add an ffs prototype to dix.h, and add includes to ffs.c.
(cherry picked from 34164e551e4c3909322d50b09835ca4ac1d49d68 commit)
---
 dix/ffs.c     |    2 ++
 include/dix.h |    3 +++
 2 files changed, 5 insertions(+), 0 deletions(-)

Index: xorg-server/dix/ffs.c
===================================================================
--- xorg-server.orig/dix/ffs.c	2006-11-13 19:59:22.000000000 +0100
+++ xorg-server/dix/ffs.c	2006-11-26 02:57:14.000000000 +0100
@@ -30,6 +30,8 @@
 ffs(int i)
 {
     int j;
+    if (i == 0)
+        return 0;
     for (j = 1; (i & 1) == 0; j++)
 	i >>= 1;
     return j;
Index: xorg-server/include/dix.h
===================================================================
--- xorg-server.orig/include/dix.h	2006-11-13 19:59:23.000000000 +0100
+++ xorg-server/include/dix.h	2006-11-26 02:57:14.000000000 +0100
@@ -815,4 +815,7 @@
 extern int xstrcasecmp(char *s1, char *s2);
 #endif
 
+/* ffs.c */
+extern int ffs(int i);
+
 #endif /* DIX_H */
